src/unitxt/inference.py

```diff
@@ -1453,27 +1456,54 @@ class OpenAiInferenceEngine(
     }
     data_classification_policy = ["public"]
     parameters: Optional[OpenAiInferenceEngineParams] = None
+    base_url: Optional[str] = None
+    default_headers: Optional[dict[str, str]] = None
+    credentials: Optional[CredentialsOpenAi] = {}
 
     def get_engine_id(self):
         return get_model_and_label_id(self.model_name, self.label)
 
```

Comment:
is this sued anymore?

Reply:
This allows for the `OpenAiInferenceEngine` to be used against any OpenAI API. We are adding a specific engine class for RITS, but users may the `OpenAiInferenceEngine` against any other OpenAI API endpoint, without needing to create a new class. For example, in the `test_rits_inference()` test, it is shown how the user can also target RITS, without using the `RITSInferenceEngine `.

Reply:
Sorry. I wasn't clear. I meant if the `def get_api_param` is needed, as I see you added the new `def _prepare_credentials(self) -> CredentialsOpenAi:`
which does te same and more

Reply:
This is used in a derived class.